src/core/server/models/user/user.ts

```diff
@@ -1781,6 +1844,55 @@ export async function banUser(
   return result.value;
 }
 
+export async function removeUserSiteBan(
+  mongo: Db,
+  tenantID: string,
+  id: string,
+  createdBy: string,
+  now = new Date(),
+  siteIDs?: string[]
+) {
+  // Create the new ban.
+  const ban: BanStatusHistory = {
+    id: uuid(),
+    active: false,
+    createdBy,
+    createdAt: now,
+    siteIDs,
+  };
+
+  // Try to update the user if the user isn't already banned.
+  const result = await collection(mongo).findOneAndUpdate(
+    {
+      id,
+      tenantID,
+    },
+    {
+      $push: {
+        "status.ban.history": ban,
+      },
```

Comment:
```suggestion
      $push: {
        "status.ban.history": ban,
      },
      $pull: {
        "status.ban.siteIDs": siteIDs,
      }
```

Reply:
In future, don't forget to include a `$in` or else it will try and delete the array of elements instead of each element and never pull anything from the array.

```
$pull: {
  "status.ban.siteIDs": { $in: siteIDs },
},
```